compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -197,6 +235,137 @@ private void performCodeGen(boolean shrink) {
         codeGenCompleted = true;
     }
 
+    private void registerUnusedBIRNodes() {
+        // Reversed the for loop because used BIRNode analysis should start from the root module.
+        // Root module is usually found last in the topologicallySortedModuleList.
+        for (int i = pkgResolution.topologicallySortedModuleList().size() - 1; i >= 0; i--) {
```

Comment:
Lets extract this `pkgResolution.topologicallySortedModuleList()` to a variable.

Reply:
Fixed in aa8254a61e2f3d88c1fdd944ffe5a5b257ede583.